static/js/components/SourceList.jsx

```diff
@@ -3,24 +3,39 @@ import PropTypes from 'prop-types';
 import { Link } from 'react-router-dom';
 
 
-const SourceList = ({ sources }) => (
-  <div>
-    <h2>
-Sources
-    </h2>
-    <ul>
-      {
-        sources.map((source, idx) => (
-          <li key={source.id}>
-            <Link to={`/source/${source.id}`}>
-              {source.id}
-            </Link>
-          </li>
-        ))
-      }
-    </ul>
-  </div>
-);
+const SourceList = ({ sources }) => {
+  if (typeof sources === 'string' || sources instanceof String) {
```

Comment:
I think we want to avoid returning varying types from our API.  Better would be to do a check for an empty list here, and then display something like "There are no sources to display."

Reply:
@stefanv this is per our conversation about needing to distinguish between the case in which there are no sources that a user has access to versus the case in which the sources table is empty. Returning an empty array doesn't help to distinguish between these two cases. Any thoughts?